applications/GeoMechanicsApplication/custom_utilities/interface_element_utilities.cpp

```diff
@@ -0,0 +1,217 @@
+    rNu(1, 10) = -Ncontainer(GPoint, 3);
+    rNu(2, 2)  = -Ncontainer(GPoint, 0);
+    rNu(2, 5)  = -Ncontainer(GPoint, 1);
+    rNu(2, 8)  = -Ncontainer(GPoint, 2);
+    rNu(2, 11) = -Ncontainer(GPoint, 3);
+
+    rNu(0, 12) = Ncontainer(GPoint, 4);
+    rNu(0, 15) = Ncontainer(GPoint, 5);
+    rNu(0, 18) = Ncontainer(GPoint, 6);
+    rNu(0, 21) = Ncontainer(GPoint, 7);
+    rNu(1, 13) = Ncontainer(GPoint, 4);
+    rNu(1, 16) = Ncontainer(GPoint, 5);
+    rNu(1, 19) = Ncontainer(GPoint, 6);
+    rNu(1, 22) = Ncontainer(GPoint, 7);
+    rNu(2, 14) = Ncontainer(GPoint, 4);
+    rNu(2, 17) = Ncontainer(GPoint, 5);
+    rNu(2, 20) = Ncontainer(GPoint, 6);
+    rNu(2, 23) = Ncontainer(GPoint, 7);
+}
+
+void InterfaceElementUtilities::FillPermeabilityMatrix(BoundedMatrix<double, 2, 2>& rPermeabilityMatrix,
+                                                       const double& JointWidth,
+                                                       const double& Transversal_Permeability)
+{
+    // Quadrilateral_interface_2d_4
+    rPermeabilityMatrix(0, 0) = JointWidth * JointWidth / 12.0;
+    rPermeabilityMatrix(1, 1) = Transversal_Permeability;
+}
+
+void InterfaceElementUtilities::FillPermeabilityMatrix(BoundedMatrix<double, 3, 3>& rPermeabilityMatrix,
+                                                       const double& JointWidth,
+                                                       const double& Transversal_Permeability)
+{
+    // Prism_interface_3d_6 and Hexahedral_interface_3d_8
+    rPermeabilityMatrix(0, 0) = JointWidth * JointWidth / 12.0;
+    rPermeabilityMatrix(1, 1) = JointWidth * JointWidth / 12.0;
+    rPermeabilityMatrix(2, 2) = Transversal_Permeability;
+}
+
+void InterfaceElementUtilities::CalculateVoigtVector(array_1d<double, 2>& rVoigtVector)
+{
+    // Quadrilateral_interface_2d_4
+    rVoigtVector[0] = 0.0;
+    rVoigtVector[1] = 1.0;
+}
+
+void InterfaceElementUtilities::CalculateVoigtVector(array_1d<double, 3>& rVoigtVector)
+{
+    // Prism_interface_3d_6 and Hexahedral_interface_3d_8
+    rVoigtVector[0] = 0.0;
+    rVoigtVector[1] = 0.0;
+    rVoigtVector[2] = 1.0;
+}
+
+void InterfaceElementUtilities::CalculateLinkPermeabilityMatrix(BoundedMatrix<double, 2, 2>& rPermeabilityMatrix,
+                                                                const double& JointWidth)
+{
+    // Quadrilateral_interface_2d_4
+    rPermeabilityMatrix(0, 0) = JointWidth * JointWidth / 12.0;
+    rPermeabilityMatrix(1, 1) = JointWidth * JointWidth / 12.0;
+}
+
+void InterfaceElementUtilities::CalculateLinkPermeabilityMatrix(BoundedMatrix<double, 3, 3>& rPermeabilityMatrix,
+                                                                const double& JointWidth)
+{
+    // Prism_interface_3d_6 and Hexahedral_interface_3d_8
+    rPermeabilityMatrix(0, 0) = JointWidth * JointWidth / 12.0;
+    rPermeabilityMatrix(1, 1) = JointWidth * JointWidth / 12.0;
+    rPermeabilityMatrix(2, 2) = JointWidth * JointWidth / 12.0;
+}
```

Comment:
These functions are just moved. I agree they should be improved and actually most of them can be removed, after we have our interface stress state, but that is out of scope for this PR in my opinion